tests/test_update_ssh_keys.rs

```diff
@@ -0,0 +1,705 @@
+
+                let authkeyone: PathBuf = PathBuf::from(format!("{}/authorized_keys.d/{}", ctx.ssh_dir.to_str().unwrap(), "two"));
+                assert!(authkeyone.exists());
+
+                let mut svec: Vec<&str> = Vec::new();
+                svec.push(sshkeyname1);
+                assert_has_keys(ctx.ssh_dir.to_str().unwrap(), svec, true);
+            }
+        }
+
+        // add two again
+        // "update-ssh-keys --add two valid2.pub"
+        //
+        // NOTE: Ideally adding two again here is supposed to immediately fail,
+        // but for some reason it simply hangs forever. For now do not run the
+        // following command to proceed to the next tests.
+
+        //        let assert_keys = ["valid1"].to_vec();
+        //        match add_one_ssh_key(
+        //            ssh_dir,      // ssh_dir
+        //            pubkeyvalid2, // pubkeyname
+        //            testnametwo,  // testname
+        //            false,        // is_force
+        //            true,         // is_replace
+        //        ) {
+        //            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid2, testnametwo, assert_keys),
+        //            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        //        }
+    }
+
+    #[test]
+    fn test_disable() {
+        let ctx = setup_tests();
+
+        run_test_disable(ctx);
+    }
+
+    #[test]
+    fn test_enable() {
+        let ctx = setup_tests();
+
+        let ssh_dir = ctx.ssh_dir.to_str().unwrap();
+        let pubkeyvalid2 = "valid2";
+        let testnametwo = "two";
+        let assert_keys = ["valid1", "valid2"].to_vec();
+
+        // "update-ssh-keys --disable two"
+        run_test_disable(ctx.clone());
+
+        // "update-ssh-keys --add-force two valid2.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid2, // pubkeyname
+            testnametwo,  // testname
+            true,         // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid2, testnametwo, assert_keys),
+            Err(err) => panic!("update_ssh_keys --add-force failed {}", err),
+        }
+    }
+
+    #[test]
+    fn test_add_bad() {
+        let ctx = setup_tests();
+
+        let ssh_dir = ctx.ssh_dir.to_str().unwrap();
+        let pubkeyvalid1 = "valid1";
+        let testnameone = "one";
+        let assert_keys = ["valid1"].to_vec();
+
+        // "update-ssh-keys --add one valid1.pub"
+        match add_one_ssh_key(
+            ssh_dir,      // ssh_dir
+            pubkeyvalid1, // pubkeyname
+            testnameone,  // testname
+            false,        // is_force
+            true,         // is_replace
+        ) {
+            Ok(keys) => add_key_check_results(ssh_dir, keys, pubkeyvalid1, testnameone, assert_keys),
+            Err(err) => panic!("update_ssh_keys --add failed {}", err),
+        }
```

Comment:
This is slightly different from what the Python test does because it doesn't call `assert_has_keys` for `valid1` here.
I suggest to call `test_add_one_file_func(ctx)` here to keep the same behavior and reuse code.

Reply:
Got it now, the `assert_has_keys` is also run. Still, `run_test_add_one_file(ctx)` would be good instead of copying the code.

Reply:
called `run_test_add_one_file()`.